source/particles/particle_handler.cc

```diff
@@ -1483,7 +1499,440 @@ namespace Particles
   }
 #endif
 
+#ifdef DEAL_II_WITH_MPI
+  template <int dim, int spacedim>
+  void
+  ParticleHandler<dim, spacedim>::send_recv_cache_particles(
```

Comment:
I thinks we might be able to merge `send_recv_cache_particles` and `send_recv_cache`.  As far as I see it, you perform in `send_recv_cache_particles` and store some additional data if the construction of the cache is requested.

Reply:
Yeah. We could definitely merge them. We could pass the boolean "build_cache" to send_receive_particles and it would take care of building the cache if needed. The added cost would be a  few extra ifs, but it would remove all code duplication. I just did not want to do this now because I was developing it and it was already a mess to understand what was happening. But definitely, now that it works and that I have the test, that could be a good idea.